geoserver/src/eumetsat/pinning/src/main/java/org/geoserver/eumetsat/pinning/ViewsEvaluator.java

```diff
@@ -0,0 +1,543 @@
+        String resetQuery = pinBatcher.resetPins(tableName);
+        logger.log(Level.FINEST, "Resetting pin query: " + resetQuery);
+    }
+
+    private void pinGeoserverLayer(
+            String layer, Instant originalTime, Instant mainTime, boolean addPin)
+            throws SQLException, IOException {
+        List<MappedLayer> mappedLayers = layersMapper.getLayersById(layer);
+        for (MappedLayer mappedLayer : mappedLayers) {
+            logger.log(Level.FINER, String.format("Updating pins for layer %s:", mappedLayer));
+            setPinLayer(originalTime, mainTime, mappedLayer, addPin);
+        }
+    }
+
+    /**
+     * Redo the view by unpinning the layers that are no longer in the view and pinning the layers
+     * that get added
+     */
+    private void redoView(ViewRecord view, ParsedView parsed) throws SQLException, IOException {
+        logger.log(
+                Level.FINE,
+                String.format(
+                        "Updating the view %d: stored view originalTime: %s, updating view time: %s",
+                        view.getId(), view.getTimeOriginal(), parsed.getTime()));
+        LayersUpdate layersUpdate = new LayersUpdate(view.getLayers(), parsed.getLayers());
+
+        // Unpin removed layers
+        updateLayersPins(
+                layersUpdate.getRemoved(), view.getTimeOriginal(), view.getTimeMain(), false);
+
+        // Pin newly added layers
+        ViewRecord updatedView = buildView(parsed);
+        updateLayersPins(layersUpdate.getAdded(), parsed.getTime(), parsed.getTime(), false);
+
+        // recompute pinning for existing layers
+        if (!updatedView.getTimeMain().equals(view.getTimeMain())
+                && !updatedView.getTimeOriginal().equals(view.getTimeOriginal())) {
+            updateLayersPins(
+                    layersUpdate.getUnchanged(), view.getTimeOriginal(), view.getTimeMain(), false);
+            updateLayersPins(
+                    layersUpdate.getUnchanged(),
+                    updatedView.getTimeOriginal(),
+                    updatedView.getTimeMain(),
+                    true);
+        }
+    }
+
+    /** Extend the pinning window of a View */
+    private void extendView(ViewRecord view, ParsedView parsed) throws SQLException, IOException {
+        logger.log(
+                Level.FINE,
+                String.format(
+                        "Updating the view %d: stored view originalTime: %s, updating view time: %s",
+                        view.getId(), view.getTimeOriginal(), parsed.getTime()));
+        LayersUpdate layersUpdate = new LayersUpdate(view.getLayers(), parsed.getLayers());
+
+        // Unpin removed layers
+        updateLayersPins(
+                layersUpdate.getRemoved(), view.getTimeOriginal(), view.getTimeMain(), false);
+
+        // Pin newly added layers
+        updateLayersPins(layersUpdate.getAdded(), view.getTimeOriginal(), parsed.getTime(), false);
```

Comment:
You say "pin newly added layers" but the last parameter of updateLayersPins is still false? Doesn't that unpin?

Reply:
You are right. there was a Typo after some recent refactoring.
Fixed.